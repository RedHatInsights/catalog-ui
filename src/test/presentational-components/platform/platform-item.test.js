import React from 'react';
import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import PlatformItem from '../../../presentational-components/platform/platform-item';

describe('<PlatformItem />', () => {
  let initialProps;
  beforeEach(() => {
    initialProps = {};
  });

  it('should render correctly', () => {
    expect(shallowToJson(shallow(<PlatformItem { ...initialProps } />))).toMatchSnapshot();
  });

  it('should call handle check callback', () => {
    const onToggleItemSelect = jest.fn();
    const wrapper = mount(<PlatformItem { ...initialProps } editMode onToggleItemSelect={ onToggleItemSelect } description="Foo" id="foo" />);
    wrapper.find('svg').last().simulate('click');
    expect(onToggleItemSelect).toHaveBeenCalled();
  });
});
