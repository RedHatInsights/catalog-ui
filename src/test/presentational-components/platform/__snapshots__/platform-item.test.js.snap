// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlatformItem /> should render correctly 1`] = `
<PlatformItem
  id="Foo"
>
  <Styled(Card)
    key="Foo"
    ouiaId="platform-itemFoo"
  >
    <Card
      className="sc-bqyKva cna-DTu"
      ouiaId="platform-itemFoo"
    >
      <article
        className="pf-c-card sc-bqyKva cna-DTu"
        data-ouia-component-id="platform-itemFoo"
        data-ouia-component-type="PF4/Card"
        data-ouia-safe={true}
        id=""
      >
        <CardHeader>
          <div
            className="pf-c-card__header"
          >
            <Styled(Level)>
              <Level
                className="sc-hBEYos iXOBOJ"
              >
                <div
                  className="pf-l-level sc-hBEYos iXOBOJ"
                >
                  <CardIcon
                    height={40}
                  >
                    <styled.div
                      height={40}
                    >
                      <div
                        className="sc-iBPRYJ dinazv"
                        height={40}
                      >
                        <IconPlaceholder
                          height={40}
                        >
                          <svg
                            height={40}
                            width={40}
                          >
                            <circle
                              cx={20}
                              cy={20}
                              fill="#ecebeb"
                              r={20}
                            />
                          </svg>
                        </IconPlaceholder>
                        <Styled(t)
                          delayTime={0}
                          height={0}
                          hidden={true}
                          onError={[Function]}
                          onLoad={[Function]}
                          src="test-file-stub"
                          threshold={2000}
                        >
                          <t
                            afterLoad={[Function]}
                            beforeLoad={[Function]}
                            className="sc-gKsewC gihfPu"
                            delayMethod="throttle"
                            delayTime={0}
                            effect=""
                            height={0}
                            hidden={true}
                            onError={[Function]}
                            onLoad={[Function]}
                            placeholderSrc={null}
                            src="test-file-stub"
                            threshold={2000}
                            useIntersectionObserver={true}
                            visibleByDefault={false}
                            wrapperClassName=""
                          >
                            <t
                              afterLoad={[Function]}
                              beforeLoad={[Function]}
                              className="sc-gKsewC gihfPu"
                              delayMethod="throttle"
                              delayTime={0}
                              height={0}
                              threshold={2000}
                              useIntersectionObserver={true}
                              visibleByDefault={false}
                            >
                              <img
                                className="sc-gKsewC gihfPu"
                                height={0}
                                hidden={true}
                                onError={[Function]}
                                onLoad={[Function]}
                                src="test-file-stub"
                              />
                            </t>
                          </t>
                        </Styled(t)>
                      </div>
                    </styled.div>
                  </CardIcon>
                </div>
              </Level>
            </Styled(Level)>
          </div>
        </CardHeader>
        <ServiceOfferingCardBody
          id="Foo"
        >
          <Styled(CardBody)>
            <CardBody
              className="sc-dQppl dRqchO"
            >
              <div
                className="pf-c-card__body sc-dQppl dRqchO"
              >
                <TextContent>
                  <div
                    className="pf-c-content"
                  >
                    <ConditionalLink>
                      <Text
                        component="h3"
                      >
                        <h3
                          className=""
                          data-pf-content={true}
                        >
                          <styled.div>
                            <div
                              className="sc-crrsfI dsScIm"
                            />
                          </styled.div>
                        </h3>
                      </Text>
                    </ConditionalLink>
                  </div>
                </TextContent>
                <ItemDetails
                  id="Foo"
                  toDisplay={
                    Array [
                      "long_description",
                    ]
                  }
                >
                  <Truncate
                    ellipsis={
                      <Tooltip
                        content={<div />}
                      >
                        <span>
                          ...
                        </span>
                      </Tooltip>
                    }
                    key="tcard-prop-long_description"
                    lines={3}
                    trimWhitespace={false}
                    width={0}
                  >
                    <span
                      width={0}
                    >
                      <span />
                      <span>
                        <styled.div
                          key="card-prop-long_description"
                        >
                          <div
                            className="sc-fubCfw ihngmA"
                          />
                        </styled.div>
                      </span>
                      <span
                        style={
                          Object {
                            "left": 0,
                            "position": "fixed",
                            "top": 0,
                            "visibility": "hidden",
                          }
                        }
                      >
                        <Tooltip
                          content={<div />}
                        >
                          <Popper
                            appendTo={[Function]}
                            distance={15}
                            enableFlip={true}
                            flipBehavior={
                              Array [
                                "top",
                                "right",
                                "bottom",
                                "left",
                                "top",
                                "right",
                                "bottom",
                              ]
                            }
                            isVisible={false}
                            onBlur={[Function]}
                            onDocumentClick={false}
                            onDocumentKeyDown={[Function]}
                            onFocus={[Function]}
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                            onTriggerEnter={[Function]}
                            placement="top"
                            popper={
                              <div
                                className="pf-c-tooltip"
                                id="pf-tooltip-1"
                                role="tooltip"
                                style={
                                  Object {
                                    "maxWidth": null,
                                    "opacity": 0,
                                    "transition": "opacity 300ms cubic-bezier(.54, 1.5, .38, 1.11)",
                                  }
                                }
                              >
                                <TooltipArrow />
                                <TooltipContent
                                  isLeftAligned={false}
                                >
                                  <div />
                                </TooltipContent>
                              </div>
                            }
                            popperMatchesTriggerWidth={false}
                            positionModifiers={
                              Object {
                                "bottom": "pf-m-bottom",
                                "left": "pf-m-left",
                                "right": "pf-m-right",
                                "top": "pf-m-top",
                              }
                            }
                            trigger={
                              <span>
                                ...
                              </span>
                            }
                            zIndex={9999}
                          >
                            <FindRefWrapper
                              onFoundRef={[Function]}
                            >
                              <span>
                                ...
                              </span>
                            </FindRefWrapper>
                          </Popper>
                        </Tooltip>
                      </span>
                    </span>
                  </Truncate>
                </ItemDetails>
              </div>
            </CardBody>
          </Styled(CardBody)>
        </ServiceOfferingCardBody>
        <CardFooter>
          <div
            className="pf-c-card__footer"
          />
        </CardFooter>
      </article>
    </Card>
  </Styled(Card)>
</PlatformItem>
`;
